nucleus/experimental/hosted_inference_client.py

```diff
@@ -0,0 +1,262 @@
+        if "signedUrl" not in model_bundle_s3_url:
+            raise Exception(
+                "Error in server request, no signedURL found"
+            )  # TODO code style broad exception
+        s3_path = model_bundle_s3_url["signedUrl"]
+        raw_s3_url = f"s3://{model_bundle_s3_url['bucket']}/{model_bundle_s3_url['key']}"
+
+        # Make bundle upload
+        bundle = dict(model=model, load_predict_fn=load_predict_fn)
+        serialized_bundle = cloudpickle.dumps(bundle)
+        requests.put(s3_path, data=serialized_bundle)
+
+        self.connection.post(
+            payload=dict(id=model_bundle_name, location=raw_s3_url),
+            route="model_bundle",
+        )  # TODO use return value somehow
+        # resp["data"]["bundle_name"] should equal model_bundle_name
+        # TODO check that a model bundle was created and no name collisions happened
+        return ModelBundle(model_bundle_name)
+
+    # TODO some function to get requirements
+
+    def create_model_endpoint(
+        self,
+        service_name: str,
+        model_bundle: ModelBundle,
+        cpus: int,
+        memory: str,
+        gpus: int,
+        min_workers: int,
+        max_workers: int,
+        per_worker: int,
+        requirements: List[str],  # Dict[str, str],
+        env_params: Dict[str, str],
+        gpu_type: Optional[str] = None,
+    ):
+        """
+        Creates a Model Endpoint that is able to serve requests
+
+        Parameters:
+            service_name: Name of model endpoint. Must be unique.
+            model_bundle: The ModelBundle that you want your Model Endpoint to serve
+            cpus: Number of cpus each worker should get, e.g. 1, 2, etc.
+            memory: Amount of memory each worker should get, e.g. "4Gi", "512Mi", etc.
+            gpus: Number of gpus each worker should get, e.g. 0, 1, etc.
+            min_workers: Minimum number of workers for model endpoint
+            max_workers: Maximum number of workers for model endpoint
+            per_worker: An autoscaling parameter. Use this to make a tradeoff between latency and costs,
+                a lower per_worker will mean more workers are created for a given workload
+            requirements: TODO we should automatically determine this. A list of python package requirements, e.g.
+                ["tensorflow==2.3.0", "tensorflow-hub==0.11.0"]
+            env_params: TODO needs more explaining. A dictionary that dictates environment information e.g.
+                the use of pytorch or tensorflow, which cuda/cudnn versions to use.
+                Specifically, the dictionary should contain the following keys:
+                "framework_type": either "tensorflow" or "pytorch".
+                "pytorch_version": Version of pytorch, e.g. "1.5.1", "1.7.0", etc. Only applicable if framework_type is pytorch
+                "cuda_version": Version of cuda used, e.g. "11.0".
```

Comment:
Heads up, I'm changing this: https://github.com/scaleapi/models/pull/2363

Though I wouldn't worry about getting all of these exactly right at this time. We can do another audit pass.